File: src/Application Specific/Classes/Entity Collections/Site/SiteCustomerAudit/SiteCustomerAudit.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace SiteCustomerAudits
    {
        public class SiteCustomerAudit
        {
            private DataTypeValidator _dataTypeValidator;

            public SiteCustomerAudit()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _SiteCustomerAuditID = 0;

            public int SiteCustomerAuditID
            {
                get
                {
                    return _SiteCustomerAuditID;
                }
            }

            public object SetSiteCustomerAuditID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SiteCustomerAuditID", value);
                }
            }

            private int _SiteID = 0;

            public int SiteID
            {
                get
                {
                    return _SiteID;
                }
            }

            public object SetSiteID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SiteID", value);
                }
            }

            private int _PreviousCustomerID = 0;

            public int PreviousCustomerID
            {
                get
                {
                    return _PreviousCustomerID;
                }
            }

            public object SetPreviousCustomerID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PreviousCustomerID", value);
                }
            }

            private DateTime _DateChanged = default;

            public DateTime DateChanged
            {
                get
                {
                    return _DateChanged;
                }

                set
                {
                    _DateChanged = value;
                }
            }

            private int _userID = 0;

            public int UserID
            {
                get
                {
                    return _userID;
                }
            }

            public object SetUserID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_userID", value);
                }
            }



            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Site/SiteCustomerAudit/SiteCustomerAudit.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace SiteCustomerAudits

        Public Class SiteCustomerAudit

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "SiteCustomerAudit Properties"


            Private _SiteCustomerAuditID As Integer = 0
            Public ReadOnly Property SiteCustomerAuditID() As Integer
                Get
                    Return _SiteCustomerAuditID
                End Get
            End Property
            Public WriteOnly Property SetSiteCustomerAuditID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SiteCustomerAuditID", Value)
                End Set
            End Property


            Private _SiteID As Integer = 0
            Public ReadOnly Property SiteID() As Integer
                Get
                    Return _SiteID
                End Get
            End Property
            Public WriteOnly Property SetSiteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SiteID", Value)
                End Set
            End Property


            Private _PreviousCustomerID As Integer = 0
            Public ReadOnly Property PreviousCustomerID() As Integer
                Get
                    Return _PreviousCustomerID
                End Get
            End Property
            Public WriteOnly Property SetPreviousCustomerID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PreviousCustomerID", Value)
                End Set
            End Property


            Private _DateChanged As DateTime = Nothing
            Public Property DateChanged() As DateTime
                Get
                    Return _DateChanged
                End Get
                Set(ByVal Value As DateTime)
                    _DateChanged = Value
                End Set
            End Property

            Private _userID As Integer = 0
            Public ReadOnly Property UserID() As Integer
                Get
                    Return _userID
                End Get
            End Property
            Public WriteOnly Property SetUserID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_userID", Value)
                End Set
            End Property



#End Region

        End Class

    End Namespace

End Namespace

